### DISCUSSION TITLE
HP reverb G2 + MRTK 2.6.1 + unity 2019.4

### DISCUSSION THREAD

**Discussion by sebstax7** [microsoft/MixedRealityToolkit-Unity#9730] [Q&A] [answered]
Dear all,

I'm trying to build a basic prototype and I don't find basic tutorial which explain the "how to" set up HP reverb G2 controllers. I mean which prefab/3D object set up in my hierarchy, which script do I add to my prefac/3D object.

I don't understand how to manage the inspector device manager.

could you please provide screenshots guideline to set up controllers? The current documentation does not help unfortunately or I missed something.

Many thanks in advance

**Comment by polar-kev** [comment, marked as answer]
Hi @sebstax7, controllers are typically detected automatically, however, for the HP Reverb G2, the Microsoft.MixedReality.Input package is required to support additional functionality of that specific controller. Here's a documentation page,
https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/supported-devices/controllers-mrtk?view=mrtkunity-2021-05#hp-reverb-g2-controllers

**Reply by sebstax7** [reply]
@polar-kev thanks

Problem solved, I confirm controllers works fine.
My mistake was I had checked "Windows Mixed Reality" in XR PLUG in management instead of "OPEN XR" and then, checked Windows Mixed reality which only appear once "OPEN XR" is checked.
